src/impl_owned_array.rs

```diff
@@ -401,16 +402,88 @@ impl<A, D> Array<A, D>
     ///            [0., 0., 0., 0.],
     ///            [1., 1., 1., 1.]]);
     /// ```
-    pub fn push(&mut self, axis: Axis, array: ArrayView<A, D::Smaller>)
-        -> Result<(), ShapeError>
+    pub fn push(&mut self, axis: Axis, array: ArrayView<A, D::Smaller>) -> Result<(), ShapeError>
     where
         A: Clone,
         D: RemoveAxis,
     {
         // same-dimensionality conversion
-        self.append(axis, array.insert_axis(axis).into_dimensionality::<D>().unwrap())
+        self.append(
+            axis,
+            array.insert_axis(axis).into_dimensionality::<D>().unwrap(),
+        )
     }
 
+    /// Return the offset between the pointer to the beginning of the heap
+    /// allocated by Array (`self.data.ptr`) and `self.ptr`.
+    unsafe fn offset_from_data(&self) -> isize {
+        if std::mem::size_of::<A>() != 0 {
+            self.as_ptr().offset_from(self.data.as_ptr())
+        } else {
+            0
+        }
+    }
+
+    /// Convert from any stride to the default stride
+    pub fn to_default_stride(&mut self) {
+        let view_mut =
```

Comment:
I'm not sure about all the details of this method, but I think it's better to take a step back and see if there isn't already some functionality that can do this - maybe .to_shape()?

Reply:
thank you for your review!

Let me check my understanding of `shrink_to_fit`.My understanding of `shrink_to_fit()` is that, for example, you have the following array, and you want to drop the 0,1-th row.

```
1 2 3 4    <- drop
5 6 7 8   <- drop
9 10 11 12
13 14 15 16
```
If you do `shrink_to_fit()` on this `Array`, I think it is supposed to change the memory order as follows.

before
```
1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16
```
after (Here, ○ is `free`ed element.)
```
9 10 11 12 13 14 15 16 ○ ○ ○ ○ ○ ○ ○ ○
```
My understanding of `shrink_to_fit` goes like this. Is it wrong?

As I understand it, `to_shape` and `into_shape` do not seem to change the memory order. Therefore, I created `to_default_stride()`. (Of course, you may not need to change it to default stride.The reason for changing stride to default stride is that we thought that the number of times the element would be swapped would not be significantly different if it were any other stride.)